docs/api.rst

```diff
@@ -0,0 +1,282 @@
+HPy API
+=======
+
+.. warning::
+   HPy is still in the early stages of development and as such the API is
+   subsequent to changes
+
+Handles
+-------
+
+The "H" in HPy stands for **handle**, which is a central concept: handles are
+used to hold a C reference to Python objects, and they are represented by the
+C ``HPy`` type.  They play the same role as ``PyObject *`` in the Python/C
+API, albeit with some important differences which are detailed below.
+
+When they are no longer needed, handles must be closed by calling
+``HPy_Close``, which plays more or less the same role as ``Py_DECREF``.
+Similarly, if you need a new handle for an existing object, you can duplicate
+it by calling ``HPy_Dup``, which plays more or less the same role as
+``Py_INCREF``.
+
+The concept of handles is certainly not unique to HPy. Other examples include
+Unix file descriptors, where you have ``dup()`` and ``close()``, and Windows'
+``HANDLE``, where you have ``DuplicateHandle()`` and ``CloseHandle()``.
+
+
+Handles vs ``PyObject *``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. XXX I don't like this sentence, but I can't come up with anything better
```

Comment:
It reads fine to me :)

I'd maybe phrase it as:

> The biggest difference is that in the old Python/C API, multiple ``PyObject
> *`` references to the same objects are completely equivalent to each other. They can be passed to Python/C API functions interchangeably _including_ memory management APIs.
> 
> In particular, it does not matter which particular reference you call ``Py_INCREF`` and ``Py_DECREF`` on, as long as the total number of increfs and decrefs to the underlying object is the same at the end of the object lifetime.

Reply:
thank you, fixed by 278ffa4